packages/botonic-plugin-ner/src/preprocess/input-generator.ts

```diff
@@ -0,0 +1,45 @@
+import { Codifier } from '@botonic/nlp/dist/preprocess/codifier'
+import {
+  PADDING_TOKEN,
+  UNKNOWN_TOKEN,
+} from '@botonic/nlp/dist/preprocess/constants'
+import { Preprocessor } from '@botonic/nlp/dist/preprocess/preprocessor'
+import { tensor } from '@tensorflow/tfjs-core/dist/ops/tensor'
+import { Tensor2D } from '@tensorflow/tfjs-core/dist/tensor'
+
+export class InputGenerator {
+  constructor(
+    readonly preprocessor: Preprocessor,
+    readonly tokenCodifier: Codifier,
+    readonly entityCodifier: Codifier
+  ) {}
+
+  generate(text: string): { sequence: string[]; input: Tensor2D } {
+    const sequence = this.processText(text)
+    const input = tensor([this.processTokens(sequence)]) as Tensor2D
+    return { sequence, input }
+  }
+
+  private processText(text: string): string[] {
+    return this.preprocessor.pad(
+      this.preprocessor.stem(
+        this.preprocessor.removeStopwords(
+          this.preprocessor.tokenize(this.preprocessor.normalize(text))
+        )
+      ),
+      PADDING_TOKEN
+    )
+  }
+
+  private maskUnknownTokens(tokens: string[]): string[] {
```

Comment:
Within a class, order the functions from top/public to down/private. maskUnknownTokens should go after processTokens